src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -192,22 +193,34 @@ void DefaultOTARequestor::OnQueryImageResponse(void * context, const QueryImageR
             ChipLogDetail(SoftwareUpdate, "Available update version %" PRIu32 " is <= current version %" PRIu32 ", update ignored",
                           update.softwareVersion, requestorCore->mCurrentVersion);
 
-            requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kIdle, OTAChangeReasonEnum::kSuccess);
             requestorCore->mOtaRequestorDriver->UpdateNotFound(UpdateNotFoundReason::kUpToDate,
                                                                System::Clock::Seconds32(response.delayedActionTime.ValueOr(0)));
+            requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kIdle, OTAChangeReasonEnum::kSuccess);
         }
 
         break;
     }
     case OTAQueryStatus::kBusy:
-        requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kDelayedOnQuery, OTAChangeReasonEnum::kDelayByProvider);
-        requestorCore->mOtaRequestorDriver->UpdateNotFound(UpdateNotFoundReason::kBusy,
-                                                           System::Clock::Seconds32(response.delayedActionTime.ValueOr(0)));
+        status = requestorCore->mOtaRequestorDriver->UpdateNotFound(
+            UpdateNotFoundReason::kBusy, System::Clock::Seconds32(response.delayedActionTime.ValueOr(0)));
+        if (status == CHIP_ERROR_MAX_RETRY_EXCEEDED)
+        {
+            // Go back to idle to start Periodic Query timer
```

Comment:
No need to mention implementation details of the driver (Periodic Query timer)

Reply:
Removed the comment.